### DISCUSSION TITLE
Best way to cut a long sentence in half?

### DISCUSSION THREAD

**Discussion by Francoyy** [snakers4/silero-vad#364] [General]
Super excited about using this new tool, there are many features that can help my subtitle generation for both English and Chinese videos!
One question, in my testing, there was about 20% of time sequences that were just very long compared to most other sequences in the SRT file.
I'm trying to address those very long sentences.
Ideally, I would like those very long sentences to be cut in half, roughly from the middle, where it makes the most sense to cut them.
I tried a different number of parameters, such as:
--segment_level true --word_level false --max_chars 56
--segment_level true --word_level false --max_chars 56 --regroup sl
--segment_level true --word_level false --max_chars 56 --regroup sp

I didn't have so much luck. One example is as such:

```
250
00:09:48,560 --> 00:09:53,040
Yes, so I think the key thing is really to maximize the

251
00:09:53,080 --> 00:09:53,440
savings
```
I hope that the sentence could have been chopped more around the middle. Is there any option that would do something like that?

One possible option in mind, is that I would add `--word_level true`, and since I have the timestamps for every word, I think I could be able to chop a long sentence in half while preserving correct timestamps. But that's not perfect, because cutting in half won't guarantee it cuts in the best place.

Thanks!

**Comment by snakers4** [comment]
Hi, this looks like an ASR question, not a VAD question.

The VAD is "stupid", i.e. it only says whether there is speech or not.

We provide the following params:

https://github.com/snakers4/silero-vad/blob/563106ef8cfac329c8be5f9c5051cd365195aff9/utils_vad.py#L164-L225
